services/paymaster/src/api.rs

```diff
@@ -230,6 +232,10 @@ async fn sponsor_and_send_handler(
     let mut transaction: VersionedTransaction = bincode::deserialize(&transaction_bytes)
         .map_err(|_| (StatusCode::BAD_REQUEST, "Failed to deserialize transaction"))?;
 
+    domain_state
+        .tolls
+        .validate_toll_payment(&transaction, &state.chain_index)
+        .await?;
```

Comment:
Similar to the recent conversation about parse, don't validate -- is there any way we can represent that this validation has happened within the type system?